waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/StaticDatabaseMappingService.java

```diff
@@ -155,6 +166,17 @@ private void validateFederatedMetastoreDatabases(List<String> mappableDatabases,
     }
   }
 
+  private List<String> applyWhitelist(List<String> allDatabases, List<String> mappedDatabases) {
+    List<String> matchedDatabases = new ArrayList<String>();
+    Whitelist whitelist = new Whitelist(mappedDatabases);
+    for (String database : allDatabases) {
```

Comment:
For another day but we could consider moving this method into the Whitelist class itself, especially if we find ourselves doing this filtering in multiple places.